src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -378,20 +377,22 @@ class npc_nerubar_victim : public CreatureScript
 
         void JustDied(Unit* killer) override
         {
-            Player* player = killer->ToPlayer();
-            if (!player)
+            if (!killer || killer->GetTypeId() != TYPEID_PLAYER)
+            {
                 return;
+            }
+
+            Player* player = killer->ToPlayer();
 
-            if (player->GetQuestStatus(QUEST_TAKEN_BY_THE_SCOURGE) == QUEST_STATUS_INCOMPLETE)
+            uint8 uiRand = urand(0, 99);
+            if (uiRand < 40)
```

Comment:
@wizzymore If we want to turn all of this to SAI via https://github.com/azerothcore/azerothcore-wotlk/issues/4084, how would we do the calculation of chance? 

If we set event_chance it could spawn multiple NPCs, peon and warrior as an example

Reply:
We can use the random actionlist call action and have 2 actionlist ( 50% chance for each ) then you set the chance to the event to occur to 80%, so you have 40% chance for each to occur

Reply:
yea but both could proc at the same time, no?

Reply:
No, SMART_ACTION_CALL_RANDOM_TIMED_ACTIONLIST as per the script and documentation:
"Will select one entry from the ones provided. 0 is ignored."
Let me show you how this would look.

Reply:
AH call random Ok, didnt know it existed :D